modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/builder/SourceSchemaDelegate.java

```diff
@@ -49,10 +48,16 @@ private void registerSourceElement(SourceModel sourceModel, String typeName, Dsl
     element.setName(dslSyntax.getElementName());
     element.setType(new QName(builder.getSchema().getTargetNamespace(), typeName));
     element.setAnnotation(builder.createDocAnnotation(sourceModel.getDescription()));
-    element.setSubstitutionGroup(MULE_ABSTRACT_MESSAGE_SOURCE);
+    element.setSubstitutionGroup(getSourceSubstitutionGroup(sourceModel));
     builder.getSchema().getSimpleTypeOrComplexTypeOrGroup().add(element);
   }
 
+  private QName getSourceSubstitutionGroup(SourceModel sourceModel) {
+    return sourceModel.getStereotype().equals(MuleStereotypeFactory.source())
```

Comment:
in which case would a source have a different stereotype?

Reply:
could be a child stereotype that inherits from Source but is more restricted. Maybe something like `PollingSource` and `ListenerSource`